File: apps/web/src/lib/validators/Embeddings.ts
import { z } from "zod";

export const embeddingsValidator = z.enum([
    "openai",
    "cohere",
    "huggingface",
    "tensorflow",
    "huggingface_instruct",
    "selfhost",
    "selfhost_huggingface",
    "selfhost_huggingface_instruct",
    "hypothetical_document"
]);
export type EmbeddingsType = z.infer<typeof embeddingsValidator>;

File: apps/api/src/solidchain/schemas/embeddings.py
from enum import Enum


class Embeddings(str, Enum):
    OPENAI = "openai"
    COHERE = "cohere"
    HUGGINGFACE = "huggingface"
    TENSORFLOW = "tensorflow"
    HUGGINGFACE_INSTRUCT = "huggingface_instruct"
    SELFHOST = "selfhost"
    SELFHOST_HUGGINGFACE = "selfhost_huggingface"
    SELFHOST_HUGGINGFACE_INSTRUCT = "selfhost_huggingface_instruct"
    HYPOTHETICAL_DOCUMENT = "hypothetical_document"
